gcloud-java-compute/src/main/java/com/google/gcloud/spi/DefaultComputeRpc.java

```diff
@@ -564,6 +567,72 @@ public Operation deleteSnapshot(String snapshot, Map<Option, ?> options) {
     }
   }
 
+  @Override
+  public Operation createImage(Image image, Map<Option, ?> options) {
+    try {
+      return compute.images()
+          .insert(this.options.projectId(), image)
+          .setFields(FIELDS.getString(options))
+          .execute();
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public Image getImage(String project, String image, Map<Option, ?> options) {
+    try {
+      return compute.images()
+          .get(project, image)
+          .setFields(FIELDS.getString(options))
+          .execute();
+    } catch (IOException ex) {
+      return nullForNotFound(ex);
+    }
+  }
+
+  @Override
+  public Tuple<String, Iterable<Image>> listImages(String project, Map<Option, ?> options) {
+    try {
+      ImageList imageList = compute.images()
+          .list(project)
+          .setFilter(FILTER.getString(options))
+          .setMaxResults(MAX_RESULTS.getLong(options))
+          .setPageToken(PAGE_TOKEN.getString(options))
+          .setFields(FIELDS.getString(options))
+          .execute();
+      Iterable<Image> images = imageList.getItems();
+      return Tuple.of(imageList.getNextPageToken(), images);
+    } catch (IOException ex) {
+      throw translate(ex);
+    }
+  }
+
+  @Override
+  public Operation deleteImage(String image, Map<Option, ?> options) {
+    try {
+      return compute.images()
+          .delete(this.options.projectId(), image)
```

Comment:
This could be a problem if we try to delete a public-image which the same name as the image in this project. No?